wagtail/admin/views/generic/models.py

```diff
@@ -186,26 +234,51 @@ def search_queryset(self, queryset):
         }
         return queryset.filter(**filters)
 
+    def _get_title_column(self, column_class=TitleColumn):
+        def title_accessor(obj):
+            draftstate_enabled = self.model and issubclass(self.model, DraftStateMixin)
+
+            if draftstate_enabled and obj.latest_revision:
+                return obj.latest_revision.object_str
+            return str(obj)
+
+        return column_class(
+            "name",
+            label=gettext_lazy("Name"),
+            accessor=title_accessor,
+            get_url=self.get_edit_url,
+        )
```

Comment:
Decided not to handle the ordering of the "Name" column in this PR. I think it's better to revisit this when we design the SnippetViewSet API.